v2/internal/testcommon/kube_per_test_context.go

```diff
@@ -420,6 +424,10 @@ func (tc *KubePerTestContext) CreateResourceAndWaitWithoutCleanup(obj client.Obj
 	tc.T.Helper()
 	gen := obj.GetGeneration()
 	tc.CreateResourceUntracked(obj)
+	// Only wait for ASO objects
+	if _, ok := obj.(genruntime.MetaObject); !ok {
+		return
+	}
```

Comment:
Ditto here.

Shouldn't we wait for the resource to be fully created, regardless of whether it's ASO or not?